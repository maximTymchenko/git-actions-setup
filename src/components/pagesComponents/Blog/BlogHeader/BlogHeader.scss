@import '../../../../styles/vars';

.blog-header {
    display: flex;
    flex-direction: column;
    align-items: center;

    &__nav {
        position: fixed;
        top: 65px;
        @include section__size;
        border-bottom: 2px solid $secondary;
        display: flex;
        justify-content: center;
        background-color: #fff;
        z-index: 100;

        @media screen and (max-width: $small__tablet) {
            overflow-x: scroll;
            width: 100%;
            max-width: 600px;
        }
    }

    &__wrapper {
        overflow: hidden;
        display: flex;
        align-items: center;
        flex-direction: column;
        text-align: center;
        position: relative;
        margin-bottom: 25px;
        padding: 60px 0 125px;

        width: 100%;
        max-width: $container__size;

        @media screen and (max-width: $small__tablet) {
            margin-top: 30px;
            padding-top: 140px;
            padding-bottom: 140px;
        }

        &::before {
            position: absolute;
            content: url('../../../../assets/img/waves.svg');
            top: 10px;
            left: -280px;
            z-index: -1;

            @media screen and (max-width: $desktop) {
                left: -320px;
            }

            @media screen and (max-width: $tablet) {
                padding: 90px 0;
                left: -380px;
                top: -120px;
            }

            @media screen and (max-width: $small__tablet) {
                content: url('../../../../assets/img/medium-waves.svg');
                left: -60px;
                top: -95px;
                transform: rotateY(180deg);
            }
        }

        &::after {
            position: absolute;
            content: url('../../../../assets/img/waves.svg');
            transform: rotateX(180deg);
            top: 10px;
            right: -280px;
            z-index: -1;

            @media screen and (max-width: $desktop) {
                right: -320px;
            }

            @media screen and (max-width: $tablet) {
                right: -350px;
                bottom: -100px;
                padding: 70px 0;
            }

            @media screen and (max-width: $small__tablet) {
                content: url('../../../../assets/img/medium-waves.svg');
                right: -50px;
                bottom: -75px;
            }
        }
    }

    &__subtitle {
        max-width: 600px;
    }

    &__form {
        display: flex;
        margin-top: 40px;
        position: relative;
        left: 10px;

        @media screen and (max-width: $small__tablet) {
            flex-direction: column;
            align-items: center;
            margin-top: 20px;
        }
    }

    &__error {
        display: block;
        margin-top: 10px;
        font-family: 'OktaNeue-Black';
        font-size: 11px;
        font-weight: 900;
        letter-spacing: 1px;
        color: $color__error;
        opacity: 0;

        &--show {
            opacity: 1;
        }
    }

    &__input {
        outline: none;
        box-shadow: 4px 4px 1px 1px rgba(57, 156, 203, 0.4);
        background-color: $background__primary;
        cursor: pointer;
        width: 350px;
        height: 40px;
        font-family: 'OktaNeue-Regular';
        font-size: 15px;
        color: $text__body;
        padding-left: 10px;
        border: 2px solid $secondary;
        margin-right: 20px;
        @include transition;

        &:focus {
            border-width: 4px;
        }

        &:hover {
            background-color: $background__secondary;
        }

        &--error {
            border: 2px solid $color__error;
            box-shadow: 4px 4px 1px 1px rgba(215, 29, 91, 0.4);
            position: relative;
        }
    }
}
