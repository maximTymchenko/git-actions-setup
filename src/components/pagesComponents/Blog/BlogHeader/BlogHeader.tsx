import React, { FC, useState } from 'react';
import './BlogHeader.scss';
import SectionTitle from '../../../common/SectionTitle';
import TextDescription from '../../../common/TextDescription';
import Button from '../../../common/Button';
import arrowRight from '../../../../assets/img/arr-right.svg';
import classNames from 'classnames';
import Nav from '../../../common/Nav';

interface IBlogHEaderProps {
    blogTitle: string;
    blogSubtitle: string;
}

const BlogHeader: FC = (props: IBlogHEaderProps) => {
    const menuLinks = {
        activeLink: null,
        links: [
            { id: 1, title: 'Latest', page: '/blog/' },
            { id: 2, title: 'Editorial', page: '/blog/editorial' },
            { id: 3, title: 'Whitepapers', page: '//' },
            { id: 4, title: 'Videos', page: '//' },
            { id: 5, title: 'Resources', page: '//' },
            {
                id: 6,
                title: 'Press center',
                page: '//',
            },
        ],
    };

    const [inputData, setInputData] = useState<string>('');
    const [emailType, setEmailType] = useState<boolean>(false);

    const getData = (val: EventTarget | null) => {
        setInputData(val.target.value);

        if (typeof inputData !== 'string') {
            setEmailType(true);
        }

        if (inputData.search(/\s/) !== -1) {
            setEmailType(true);
        }

        if (inputData.search(/[!#$%^&*()\-=_+~[\]{}'"\\|,/<>?]/) === -1) {
            setEmailType(true);
        }

        if (inputData == '') {
            setEmailType(false);
        }
    };

    return (
        <div data-testid="blog-header" className="blog-header">
            <div className="blog-header__nav">
                <Nav links={menuLinks} linkType="link__blog" />
            </div>
            <div className="blog-header__wrapper">
                <SectionTitle type="section" title={props.blogTitle} />
                <div className="blog-header__subtitle">
                    <TextDescription>{props.blogSubtitle}</TextDescription>
                </div>
            </div>
            <form className="blog-header__form">
                <div className="blog-header__input-wrapper">
                    <input
                        onChange={getData}
                        className={classNames('blog-header__input', {
                            'blog-header__input--error': emailType,
                        })}
                        type="email"
                        name="email"
                        placeholder="Your email"
                    />
                    <span
                        className={classNames('blog-header__error', {
                            'blog-header__error--show': emailType,
                        })}
                    >
                        Check email
                    </span>
                </div>
                <div className="blog-header__button">
                    <Button icon={arrowRight} type="secondary">
                        Subscribe
                    </Button>
                </div>
            </form>
        </div>
    );
};

export default BlogHeader;
