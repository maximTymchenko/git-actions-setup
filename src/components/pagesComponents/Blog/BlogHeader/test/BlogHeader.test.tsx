import React from 'react';
import BlogHeader from '../BlogHeader';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

it('renders elem correctly', () => {
    const { getByTestId } = render(
        <BlogHeader
            blogTitle="Blog"
            blogSubtitle=" Articles, stories, case studies technical guides, best practices, tech videos,
                    and lots of resources to help you build, grow, and skyrocket your product or
                    digital side of your business."
        />,
    );
    expect(getByTestId('blog-header')).toBeTruthy();
});

it('renders without unexpected changes', () => {
    const tree = renderer
        .create(
            <BlogHeader
                blogTitle="Blog"
                blogSubtitle=" Articles, stories, case studies technical guides, best practices, tech videos,
                    and lots of resources to help you build, grow, and skyrocket your product or
                    digital side of your business."
            />,
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
