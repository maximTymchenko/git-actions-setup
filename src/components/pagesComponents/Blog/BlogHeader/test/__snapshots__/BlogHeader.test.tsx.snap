// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="blog-header"
  data-testid="blog-header"
>
  <div
    className="blog-header__nav"
  >
    <ul
      className="nav nav--blog"
    >
      <div
        className="nav__container nav__container--undefined"
      >
        <a
          className="link__margin  active__border nav__link link__blog"
          data-testid="nav"
          href="/blog/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <li
            className="nav__link"
            onClick={[Function]}
          >
            Latest
          </li>
        </a>
        <a
          className="link__margin  active__border nav__link link__blog"
          data-testid="nav"
          href="/blog/editorial"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <li
            className="nav__link"
            onClick={[Function]}
          >
            Editorial
          </li>
        </a>
        <a
          className="link__margin  active__border nav__link link__blog"
          data-testid="nav"
          href="/blog/whitepapers"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <li
            className="nav__link"
            onClick={[Function]}
          >
            Whitepapers
          </li>
        </a>
        <a
          className="link__margin  active__border nav__link link__blog"
          data-testid="nav"
          href="/blog/videos"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <li
            className="nav__link"
            onClick={[Function]}
          >
            Videos
          </li>
        </a>
        <a
          className="link__margin  active__border nav__link link__blog"
          data-testid="nav"
          href="/blog/resources"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <li
            className="nav__link"
            onClick={[Function]}
          >
            Resources
          </li>
        </a>
        <a
          className="link__margin  active__border nav__link link__blog"
          data-testid="nav"
          href="/blog/press-center"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <li
            className="nav__link"
            onClick={[Function]}
          >
            Press center
          </li>
        </a>
      </div>
    </ul>
  </div>
  <div
    className="blog-header__wrapper"
  >
    <h2
      className="title title--section"
      data-testid="section-title"
    >
      Blog
    </h2>
    <div
      className="blog-header__subtitle"
    >
      <p
        className="text-description text-description--undefined text-description--undefined"
        data-testid="text-description"
      >
         Articles, stories, case studies technical guides, best practices, tech videos, and lots of resources to help you build, grow, and skyrocket your product or digital side of your business.
      </p>
    </div>
    <form
      className="blog-header__form"
      data-testid="blog-form"
      onSubmit={[Function]}
    >
      <div
        className="blog-header__input-wrapper"
      >
        <input
          className="blog-header__input"
          name="email"
          placeholder="Your email"
          type="text"
        />
        <span
          className="blog-header__error"
        >
          Check email
        </span>
      </div>
      <div
        className="blog-header__button"
      >
        <button
          className="button button--secondary button--undefined button--undefined"
          data-testid="button"
        >
          Subscribe
          <img
            className="button__icon button__icon--undefined"
            data-testid="button__icon"
            src="test-file-stub"
          />
        </button>
      </div>
    </form>
  </div>
</div>
`;
